[Profile] Profile usability/consistency updates.

78andyp opened the pull request:
## Description

Provides a number of usability/consistency fixes for profiles when a remote database is in use v. profiles when local databases are in use.

Currently local profiles behave as described in the wiki - https://kodi.wiki/view/Profiles. Remote profiles do not (and this is not made clear).

The only (vague) reference to making remote database profiles work correctly is in https://kodi.wiki/view/MySQL/Setting_up_Kodi#Name_tag.

There has been discussion in the links below as to whether the should be one music and video database (which is what PR #24383 attempted) or seperate ones (which seems to be by design for local databases). I accept that using a remote database is exprimental (and needs `advancessettings.xml` knowledge - although that is well explained in the wiki) but my feeling is that profilies should behave consistently, hence this PR. I know this may all change in v22 but these relatively simple fixes will make it consistent for now.

The remainder are some usability fixes, as a result from playing around with profiles.

## Motivation and context

As highlighted in #24424 and following on from #23999 (and closed PR #24383).

1) In the profile dialog it refers to 'Media Info' - it seems to mean the actual media itself (ie. the content of the myvideos/mymusic databases) - I think it should just say 'Media'??

Changed to just say Media - more accurately reflects effect.

2) Once a profile is created and you make changes to the profile you need to log off and on again for them to take effect - this isn't explain in the GUI - perhaps a dialog offering to log off now if a change is made??

The user is now presented with a dialog to reload profile.

3) Versions and extras can still be manipulated if 'Shares with Source (Read Only)' is chosen for Media - these buttons in the info dialog should be greyed out like Change Art is.

These two buttons are now appropriately greyed out.

4) When creating a profile with a remote database - Media Info effectively doesn't do anything. The same mymovies and mymusic tables are used for each profile - this can be overcome with the tag in advancedsettings.xml but this isn't obvious - I wonder if it should be default that an advancedsettings.xml is created in the profile directory with the and tags copied over and inserted??

A custom `advancedsettings.xml` is now created in the profile directory if there is a remote database in the current profile - these details are copied over and name for the music/video database is given as <profilename>-mymusic/myvideos.

5) Once a exists for a remote database in a profile advancedsettings.xml - the media option to share default database is ignored.

This is fixed.

## How has this been tested?

Locally on Win64 with a remote mariadb database.

## What is the effect on users?

As above

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Improvement** (non-breaking change which improves existing functionality))

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [X] All new and existing tests passed

DaVukovic commented:
Just to mention it:

Regarding: 
>When profile list is displayed it's not clear which is currently active.

I wrote that for Estuary as a setting for the skin: 

https://github.com/xbmc/xbmc/pull/14007

It's not a highlight on the settings window, but at least one is able to enable a setting which shows the current profile name or avatar.

78andyp commented:
> Just to mention it:
> 
> Regarding:
> 
> > When profile list is displayed it's not clear which is currently active.
> 
> I wrote that for Estuary as a setting for the skin:
> 
> #14007
> 
> It's not a highlight on the settings window, but at least one is able to enable a setting which shows the current profile name or avatar.

Thank you. I hadn't seen that. All this PR does is:

<img width="1280" alt="image" src="https://github.com/xbmc/xbmc/assets/99039295/179b98a2-ece3-491e-9df8-7ac873bd3877">

DaVukovic commented:
I understood and it's a cool addition. 

Just want to let you know that we already included at least something to identify the profile in use ;)

78andyp committed:
[Profile] Change "Media Info" to "Media"

In the profile dialog it refers to 'Media Info' - it seems to mean the actual media itself (ie. the content of the myvideos/mymusic databases)

78andyp committed:
[Profile] Present user with option to reload profile after changes (so they take effect).

Once a profile is created and you make changes to the profile you need to log off and on again for them to take effect - this isn't explained in the GUI.

78andyp committed:
[Profile] Disable versions and extras buttons in the information dialog if the source is read only.

Versions and extras can still be manipulated if 'Shares with Source (Read Only)' is chosen for Media - these buttons in the info dialog should be greyed out like Change Art is.

78andyp committed:
[Profile] Fix option to share default database is ignored.

Once there is a remote database in a profile advancedsettings.xml the media option to share default database is ignored.

78andyp committed:
[Profile] A custom advancedsettings.xml is now created in the profile directory if there is a remote database in the current profile - these details are copied over and name for the music/video database is given as -mymusic/myvideos.

When creating a profile with a remote database - Media Info effectively doesn't do anything. The same mymovies and mymusic tables are used for each profile - this can be overcome with the tag in advancedsettings.xml but this isn't obvious.
